### PR TITLE
Support indexable selectors

### PR DESCRIPTION
**Pull request by juanazam**
Implement index option on attributes and queries to make definitions more dev friendly.

### CODE REVIEW

**Comment by san650** [comment]
This is great!

I think we can support the `index` option in all attributes, not only on the query ones. Would you mind adding the support for the rest?

If we resolve the selector inside `qualifySelector` helper then the option is available for all helpers.

File: tests/unit/queries/attribute-test.js [hunk 1]
```diff
@@ -55,3 +55,11 @@ it('uses page scope', function(assert) {
 
   assert.equal(attr(), 'Logo small');
 });
+
+it('searches for element by index if provided', function(assert) {
+  fixture('<img alt="img1" class="img"></img><img alt="img2" class="img"></img>');
```
Comment (san650) [inline tests/unit/queries/attribute-test.js]: `` tags doesn't support child elements. Would you mind changing the fixture tu use `<span>` elements instead?

File: test-support/page-object/queries.js [hunk 2]
```diff
@@ -2,11 +2,23 @@
 
 import { qualifySelector, trim } from './helpers';
 
+function indexedSelector(selector, index) {
+  var iSelector;
```
Comment (san650) [inline test-support/page-object/queries.js]: `iSelector` :crying_cat_face:

Maybe we could change `selector` to `baseSelector` and `iSelector` to `selector`?

Comment (juanazam) [inline test-support/page-object/queries.js]: :+1: I wasn't inspired while thinking of that variable name :)

File: tests/unit/queries/attribute-test.js (hunk 1 above)
Comment (juanazam) [inline tests/unit/queries/attribute-test.js]: I'm not sure if I follow, you don't have to use nested child elements to use the index options, its just a way to select from any matched elements by a certain selector.

**Comment by juanazam** [comment]
Thanks for your reply!

I didn't do it there because of the signature of that method (https://github.com/san650/ember-cli-page-object/blob/master/test-support/page-object/helpers.js#L3).

I wasn't sure if I should change the signature to something like:
```js
function qualifySelector(selector, ...scopes)
```
or assume that the last element of the parameters on the current signature is the one the index rule should apply (I think this is more error prone), which do you think is the best way?

Cheers!

File: tests/unit/queries/attribute-test.js (hunk 1 above)
Comment (san650) [inline tests/unit/queries/attribute-test.js]: What I meant was that "</img>" is invalid. It should be "".

**Comment by san650** [comment]
Maybe `function qualifySelector(selector, scope, options = {})`?

**Comment by juanazam** [comment]
Doing that would reduce how much the function can do:
right now it supports calls like: `qualifySelector('.my-outer-div', '.my-inner-div', '.my-element')`

In order to accomplish something similar with that signature: `qualifySelector('.my-outer-div .my-inner-div', '.my-element')`

With literals is not as troublesome as with variables, in that case, the call would look something like this:
`qualifySelector("${outerDiv}  ${innerDiv}", '.my-element')`

If we are ok with this, I'll implement it this way.

**Comment by san650** [comment]
+2

**Comment by san650** [comment]
Please, rebase. I've refactored a bit the code, so you may need to update your changes.

File: tests/unit/queries/attribute-test.js (hunk 1 above)
Comment (juanazam) [inline tests/unit/queries/attribute-test.js]: :+1:

**Comment by juanazam** [comment]
Rebased and attended comments :smile:

**Comment by san650** [comment]
:heart:

**Comment by juanazam** [comment]
:smile: :smile: :smile: :smile: :smile: :smile:

**Pull request merged**